core/state_processor.go

```diff
@@ -80,6 +81,9 @@ func (p *StateProcessor) Process(block *types.Block, statedb *state.StateDB, cfg
 	if beaconRoot := block.BeaconRoot(); beaconRoot != nil {
 		ProcessBeaconBlockRoot(*beaconRoot, vmenv, statedb)
 	}
+	if p.config.IsPrague(block.Number(), block.Time()) {
+		ProcessBlockHashHistory(statedb, block.Header(), p.config, p.bc)
+	}
```

Comment:
This logic is a bit convoluted. There are basically two paths: 
1. On the exact fork-block: we need to iterate and add 256 hashes
2. On every block after the fork: we need to add the parent hash. 

The code, as written, makes path `1` the 'default-path'. We always enter the `ProcessBlockHashHistory`, which invokes `ProcessParentBlockHash` and just exits before iterating. IMO it would be more intuitive to default to path `2`, invoke `ProcessParentBlockHash` by default, and only ever once invoke `ProcessBlockHashHistory`.

Reply:
Think of it this way: the parent's block hash is *always* inserted. The other 8191 block hashes are only inserted in a certain condition after the parent one. Also having one wrapping method makes the footprint inside state_processor, miner etc. smaller.